components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/model/SAMLSSOServiceProviderDO.java

```diff
@@ -154,7 +158,22 @@ public String getAttributeConsumingServiceIndex() {
     public void setAttributeConsumingServiceIndex(String attributeConsumingServiceIndex) {
         this.attributeConsumingServiceIndex = attributeConsumingServiceIndex;
     }
+    public String getSigningAlgorithm() {
+        if (StringUtils.isBlank(signingAlgorithm)) {
+            if (IdentityUtil.getProperty(IdentityConstants.ServerConfig.SSO_DEFAULT_SIGNING_ALGORITHM) != null &&
```

Comment:
replace with isNotBlank(),
move set default signing algorithm to constructor, in get method return the signingAlgorithm